libraries/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -299,12 +300,19 @@ export class RushProjectConfiguration {
    * Examines the list of source files for the project and the target phase and returns a reason
    * why the project cannot enable the build cache for that phase, or undefined if it is safe to so do.
    */
-  public getCacheDisabledReason(trackedFileNames: Iterable<string>, phaseName: string): string | undefined {
+  public getCacheDisabledReason(
+    trackedFileNames: Iterable<string>,
+    phaseName: string,
+    operation?: Operation
+  ): string | undefined {
     if (this.disableBuildCacheForProject) {
       return 'Caching has been disabled for this project.';
     }
 
-    const normalizedProjectRelativeFolder: string = Path.convertToSlashes(this.project.projectRelativeFolder);
+    // Skip no-op operations as they won't have any output/cacheable things.
+    if (operation?.runner?.isNoOp) {
+      return undefined;
+    }
```

Comment:
We should actually check this first; if the operation is a noop it doesn't matter if it is marked as uncacheable or not.